File: mortgage.py
from sys import stdin,stdout

while True:
    X,Y,N,r = stdin.readline().split()
    if X == Y == N == r == "0":
        break
    X,Y,N,r = float(X),float(Y),int(N),float(r) 
    #calculate amount of money remaining at end of nth month
    bal = X * pow(1 + (r / 1200), N * 12) - (1200 * Y * (pow(1 + (r / 1200), N * 12) - 1))/r if r != 0 else X - 12 * N * Y
    stdout.write("YES\n") if bal <= 0 else stdout.write("NO\n")

File: mortgage.cpp
#include <bits/stdc++.h>

using namespace std;

double X,Y,N,r,b;

int main() {
    while (true) {
        scanf("%lf %lf %lf %lf",&X,&Y,&N,&r);
        if (X == 0 && Y == 0 && N == 0 && r == 0) break;
        b = (r != 0 ? X * pow(1 + (r / 1200), N * 12) - (1200 * Y * (pow(1 + (r / 1200), N * 12)  - 1)) / r: X - 12 * N * Y);
        printf(b <= 0 ? "YES\n": "NO\n");
    }
    return 0;
}